loader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/loader/LoaderRow.scala

```diff
@@ -139,10 +134,7 @@ object LoaderRow {
     */
   def transformJson(resolver: Resolver[Id], schemaKey: SchemaKey)
                    (data: Json): ValidatedNel[FailureDetails.LoaderIgluError, Row] =
-    resolver.lookupSchema(schemaKey)
-      .leftMap { e => NonEmptyList.one(FailureDetails.LoaderIgluError.IgluError(schemaKey, e)) }
-      .flatMap(schema => DdlSchema.parse(schema).toRight(invalidSchema(schemaKey)))
-      .map(schema => Field.build("", schema, false))
+    LoaderCache.getOrLookup(schemaKey)(lookup(resolver, schemaKey))
```

Comment:
Snowplow LRU map seemed like an overkill for this simple use case. I see your point about caching errors! I'll add a TTL to it. 

(Btw, does LRU map have a ttl? If the schema whose error was cached is used very often, then wouldn't it stick around in the LRU map for much longer than with ttl?)

Reply:
Oops my mistake sorry, the LRU map doesn't have a TTL and wouldn't work in our case, for the exact reason that we mention. So what we really need is a TTL.

Is the same cache instance used for the whole duration of the BQ loader?

Reply:
Yes, although I think each worker will have their own instance.

I don't expect OOM to be an issue, because we're realistically only dealing with a few hundreds of schemas in the most extreme cases probably... WDYT?

Reply:
Indeed the OOM is very unlikely. A cached errored schema is still a problem IMO. What about adding a schema to the cache *only* if it is valid, and then it stays there ?

Reply:
Agree with everything above (TTL is a good idea, OOM is unlikely) and seems we already addressed it?

I wouldn't be too concerned about cached error though because this is what Iglu Client's cache deals with (so we're working with two-layered cache here). In other words if Iglu client returned an error - it might be either cached error and new error and its up to the iglu client to decide (this can be overwritten in configuration).